hypershift-operator/controllers/nodepool/capi.go

```diff
@@ -1375,3 +1316,22 @@ func (r *NodePoolReconciler) getMachinesForNodePool(ctx context.Context, nodePoo
 
 	return sortedByCreationTimestamp(machinesForNodePool), nil
 }
+
+func (r *NodePoolReconciler) getMachineSet(ctx context.Context, nodePool *hyperv1.NodePool) (*capiv1.MachineSet, error) {
+	machineSets := &capiv1.MachineSetList{}
+
+	if err := r.List(ctx, machineSets,
+		client.InNamespace(nodePool.Namespace),
+		client.MatchingLabels{
+			"cluster.x-k8s.io/deployment-name": nodePool.Name,
```

Comment:
we are only interested in machineSets for in place, in that scenario there are no machineDeployments.
You can get the machineSet for a NodePool like
	return &capiv1.MachineSet{
		ObjectMeta: metav1.ObjectMeta{
			Name:      c.nodePool.GetName(),
			Namespace: c.controlplaneNamespace,
		},
	}

See func (c *CAPI) machineSet() *capiv1.MachineSet {

Reply:
done